src/frontend/src/pages/FinancePage/RefundsSection.tsx

```diff
@@ -1,133 +1,133 @@
-              </TableRow>
-            </TableHead>
-            <TableBody>
-              {rows.map((row, index) => (
-                <TableRow
-                  key={`${row.date}-$${row.amount}-${index}`}
-                  sx={{ '&:last-child td, &:last-child th': { border: 0 } }}
-                >
-                  <TableCell align="center">{datePipe(row.date)}</TableCell>
-                  <TableCell align="center">{centsToDollar(row.amount)}</TableCell>
-                  {tabValue === 1 && <TableCell align="center">{fullNamePipe(row.recipient)}</TableCell>}
-                </TableRow>
-              ))}
-            </TableBody>
-          </Table>
-        </TableContainer>
-      </Box>
-    </Box>
-  );
-};
-
-export default Refunds;
+import {
+  Box,
+  Paper,
+  Stack,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  Typography,
+  useTheme
+} from '@mui/material';
+import { useState } from 'react';
+import { useAllReimbursements, useCurrentUserReimbursements } from '../../hooks/finance.hooks';
+import ErrorPage from '../ErrorPage';
+import LoadingIndicator from '../../components/LoadingIndicator';
+import { Reimbursement, ReimbursementRequest, isAdmin } from 'shared';
+import { useCurrentUser } from '../../hooks/users.hooks';
+import { centsToDollar, datePipe, fullNamePipe } from '../../utils/pipes';
+import NERProgressBar from '../../components/NERProgressBar';
+import ColumnHeader from './FinanceComponents/ColumnHeader';
+import FinanceTabs from './FinanceComponents/FinanceTabs';
+import { getRefundRowData } from '../../utils/reimbursement-request.utils';
+
+const RefundHeader = ({ header, data }: { header: string; data: string }) => {
+  return (
+    <Stack sx={{ alignItems: 'center' }}>
+      <Typography textAlign="center" sx={{ fontWeight: 700 }}>
+        {header}
+      </Typography>
+      <Typography> {data}</Typography>
+    </Stack>
+  );
+};
+
+interface RefundTableProps {
+  userReimbursementRequests: ReimbursementRequest[];
+  allReimbursementRequests?: ReimbursementRequest[];
+}
+
+const Refunds = ({ userReimbursementRequests, allReimbursementRequests }: RefundTableProps) => {
+  const [tabValue, setTabValue] = useState(0);
+  const user = useCurrentUser();
+
+  const {
+    data: userReimbursements,
+    isLoading: userReimbursementsIsLoading,
+    isError: userReimbursementsIsError,
+    error: userReimbursementError
+  } = useCurrentUserReimbursements();
+  const {
+    data: allReimbursements,
+    isLoading: allReimbursementsIsLoading,
+    isError: allReimbursementsIsError,
+    error: allReimbursementsError
+  } = useAllReimbursements();
+  const theme = useTheme();
+
+  if (user.isFinance && allReimbursementsIsError) return <ErrorPage message={allReimbursementsError?.message} />;
+  if (userReimbursementsIsError) return <ErrorPage message={userReimbursementError?.message} />;
+  if (
+    (user.isFinance && (allReimbursementsIsLoading || !allReimbursements)) ||

```

Comment:
you should update it in these two places as well. It's probably worth making a "hasPerms" variable and calling that everywhere user.isFinance is used rn

Reply:
When I define it there it gives the error page popup whenever I save the file and then if I switch tabs (on the dashboard and go back its fine) so I'll change it but I don't know what this behavior is

Reply:
If its clark kent its fine since he's on the finance team but if its thomas then there is weird behavior

Reply:
![image](https://github.com/Northeastern-Electric-Racing/FinishLine/assets/124754518/2750a8f7-d96c-44de-92c9-6fd1e0b2881e)

Reply:
OH nvm I think because I haven't implemented the backend checks its breaking